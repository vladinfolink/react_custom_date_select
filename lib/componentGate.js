"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var CustomDropZone_1 = require("./components/react-custom-dropzone/containers/CustomDropZone/CustomDropZone");
exports.CustomDropZone = CustomDropZone_1.CustomDropZone;
//# sourceMappingURL=componentGate.js.map