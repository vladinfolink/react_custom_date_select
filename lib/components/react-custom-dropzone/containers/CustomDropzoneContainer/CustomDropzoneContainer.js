"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var React = tslib_1.__importStar(require("react"));
var DragTarget_1 = require("../DragTarget/DragTarget");
var CustomDropZoneContainer = (function (_super) {
    tslib_1.__extends(CustomDropZoneContainer, _super);
    function CustomDropZoneContainer(props) {
        var _this = _super.call(this, props) || this;
        _this.state = {
            userIsDragging: false,
            dragCounter: 0
        };
        _this.handleDrag = _this.handleDrag.bind(_this);
        _this.handleDragIn = _this.handleDragIn.bind(_this);
        _this.handleDragOut = _this.handleDragOut.bind(_this);
        _this.handleDrop = _this.handleDrop.bind(_this);
        return _this;
    }
    CustomDropZoneContainer.prototype.handleDrag = function (event) {
        event.preventDefault();
        event.stopPropagation();
    };
    CustomDropZoneContainer.prototype.handleDragIn = function (event) {
        event.preventDefault();
        event.stopPropagation();
        this.setState(function (state) {
            return {
                dragCounter: state.dragCounter + 1
            };
        });
        if (event.dataTransfer.items) {
            this.setState(function () {
                return { userIsDragging: true };
            });
        }
    };
    CustomDropZoneContainer.prototype.handleDragOut = function (event) {
        event.preventDefault();
        event.stopPropagation();
        this.setState(function () {
            return {
                dragCounter: 0,
                userIsDragging: false
            };
        });
        if (this.state.dragCounter === 0) {
            this.setState(function () {
                return { userIsDragging: false };
            });
        }
    };
    CustomDropZoneContainer.prototype.handleDrop = function (event) {
        event.preventDefault();
        event.stopPropagation();
        var files = (event.dataTransfer || event.target).files;
        if (files) {
            this.props.handleDrop(files);
            if (event.dataTransfer)
                event.dataTransfer.clearData();
            this.setState(function (state) {
                return {
                    dragCounter: 0,
                    userIsDragging: false
                };
            });
        }
    };
    CustomDropZoneContainer.prototype.render = function () {
        var dropText = this.props.dropText;
        var userIsDragging = this.state.userIsDragging;
        return (React.createElement(DragTarget_1.DragTarget, { userIsDragging: userIsDragging, allowClick: this.props.allowClick, handleDragIn: this.handleDragIn, handleDragOut: this.handleDragOut, handleDrag: this.handleDrag, handleDrop: this.handleDrop, dropText: dropText }, this.props.children));
    };
    return CustomDropZoneContainer;
}(React.Component));
exports.CustomDropZoneContainer = CustomDropZoneContainer;
//# sourceMappingURL=CustomDropzoneContainer.js.map