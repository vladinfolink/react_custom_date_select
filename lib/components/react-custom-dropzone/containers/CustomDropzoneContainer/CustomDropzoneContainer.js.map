{"version":3,"file":"CustomDropzoneContainer.js","sourceRoot":"","sources":["../../../../../src/components/react-custom-dropzone/containers/CustomDropzoneContainer/CustomDropzoneContainer.tsx"],"names":[],"mappings":";;;AAAA,mDAA8B;AAE9B,uDAAsD;AAYtD;IAA8C,mDAA6B;IACxE,iCAAY,KAAU;QAAtB,YACC,kBAAM,KAAK,CAAC,SASb;QARC,KAAI,CAAC,KAAK,GAAG;YACX,cAAc,EAAE,KAAK;YACrB,WAAW,EAAE,CAAC;SACf,CAAC;QACF,KAAI,CAAC,UAAU,GAAG,KAAI,CAAC,UAAU,CAAC,IAAI,CAAC,KAAI,CAAC,CAAC;QAC7C,KAAI,CAAC,YAAY,GAAG,KAAI,CAAC,YAAY,CAAC,IAAI,CAAC,KAAI,CAAC,CAAC;QACjD,KAAI,CAAC,aAAa,GAAG,KAAI,CAAC,aAAa,CAAC,IAAI,CAAC,KAAI,CAAC,CAAC;QACnD,KAAI,CAAC,UAAU,GAAG,KAAI,CAAC,UAAU,CAAC,IAAI,CAAC,KAAI,CAAC,CAAC;;IAC/C,CAAC;IAEM,4CAAU,GAAlB,UAAmB,KAAU;QAC1B,KAAK,CAAC,cAAc,EAAE,CAAC;QACvB,KAAK,CAAC,eAAe,EAAE,CAAC;IAC1B,CAAC;IACM,8CAAY,GAApB,UAAqB,KAAU;QAC5B,KAAK,CAAC,cAAc,EAAE,CAAC;QACvB,KAAK,CAAC,eAAe,EAAE,CAAC;QACxB,IAAI,CAAC,QAAQ,CAAC,UAAS,KAAK;YAC1B,OAAO;gBACL,WAAW,EAAE,KAAK,CAAC,WAAW,GAAG,CAAC;aACnC,CAAC;QACJ,CAAC,CAAC,CAAC;QAEH,IAAI,KAAK,CAAC,YAAY,CAAC,KAAK,EAAE;YAC5B,IAAI,CAAC,QAAQ,CAAC;gBACZ,OAAO,EAAE,cAAc,EAAE,IAAI,EAAE,CAAC;YAClC,CAAC,CAAC,CAAC;SACJ;IACH,CAAC;IACM,+CAAa,GAArB,UAAsB,KAAU;QAC7B,KAAK,CAAC,cAAc,EAAE,CAAC;QACvB,KAAK,CAAC,eAAe,EAAE,CAAC;QACxB,IAAI,CAAC,QAAQ,CAAC;YACZ,OAAO;gBACL,WAAW,EAAE,CAAC;gBACd,cAAc,EAAE,KAAK;aACtB,CAAC;QACJ,CAAC,CAAC,CAAC;QACH,IAAI,IAAI,CAAC,KAAK,CAAC,WAAW,KAAK,CAAC,EAAE;YAChC,IAAI,CAAC,QAAQ,CAAC;gBACZ,OAAO,EAAE,cAAc,EAAE,KAAK,EAAE,CAAC;YACnC,CAAC,CAAC,CAAC;SACJ;IACH,CAAC;IACM,4CAAU,GAAnB,UAAoB,KAAU;QAC1B,KAAK,CAAC,cAAc,EAAE,CAAC;QACvB,KAAK,CAAC,eAAe,EAAE,CAAC;QAChB,IAAA,kDAAK,CAAwC;QACrD,IAAI,KAAK,EAAE;YACT,IAAI,CAAC,KAAK,CAAC,UAAU,CAAC,KAAK,CAAC,CAAC;YAC7B,IAAI,KAAK,CAAC,YAAY;gBAAE,KAAK,CAAC,YAAY,CAAC,SAAS,EAAE,CAAC;YACvD,IAAI,CAAC,QAAQ,CAAC,UAAS,KAAK;gBAC1B,OAAO;oBACL,WAAW,EAAE,CAAC;oBACd,cAAc,EAAE,KAAK;iBACtB,CAAC;YACJ,CAAC,CAAC,CAAC;SACJ;IACH,CAAC;IAED,wCAAM,GAAN;QACU,IAAA,8BAAQ,CAAgB;QACxB,IAAA,0CAAc,CAAgB;QAEtC,OAAO,CACL,oBAAC,uBAAU,IACT,cAAc,EAAE,cAAc,EAC9B,UAAU,EAAE,IAAI,CAAC,KAAK,CAAC,UAAU,EACjC,YAAY,EAAE,IAAI,CAAC,YAAY,EAC/B,aAAa,EAAE,IAAI,CAAC,aAAa,EACjC,UAAU,EAAE,IAAI,CAAC,UAAU,EAC3B,UAAU,EAAE,IAAI,CAAC,UAAU,EAC3B,QAAQ,EAAE,QAAQ,IAMjB,IAAI,CAAC,KAAK,CAAC,QAAQ,CACT,CACd,CAAC;IACJ,CAAC;IACH,8BAAC;AAAD,CAAC,AArFD,CAA8C,KAAK,CAAC,SAAS,GAqF5D;AArFa,0DAAuB"}