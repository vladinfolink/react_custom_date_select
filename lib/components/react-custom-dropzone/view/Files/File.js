"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var React = tslib_1.__importStar(require("react"));
var File = (function (_super) {
    tslib_1.__extends(File, _super);
    function File(props) {
        var _this = _super.call(this, props) || this;
        _this.state = {
            imageSource: ''
        };
        _this.doesReturnImage = _this.doesReturnImage.bind(_this);
        _this.handleImage = _this.handleImage.bind(_this);
        return _this;
    }
    File.prototype.componentDidMount = function () {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                if (this.props.renderPics)
                    this.handleImage();
                return [2];
            });
        });
    };
    File.prototype.doesReturnImage = function () {
        return this.props.file.type.split("/")[0] === "image";
    };
    File.prototype.handleImage = function () {
        var _this = this;
        (function () { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            var reader;
            var _this = this;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        reader = new window.FileReader();
                        return [4, reader.readAsDataURL(this.props.file)];
                    case 1:
                        _a.sent();
                        reader.onload = function () {
                            _this.setState(function () {
                                var result = reader.result;
                                return {
                                    imageSource: result
                                };
                            });
                        };
                        return [2];
                }
            });
        }); })();
    };
    File.prototype.render = function () {
        var _this = this;
        return (React.createElement(React.Fragment, null,
            this.props.renderPics && this.doesReturnImage() && (React.createElement("img", { className: "rcd_imgUnit", src: this.state.imageSource, alt: this.props.file.name })),
            React.createElement("span", { className: "rcd_fileNameContent" }, this.props.file.name),
            React.createElement("button", { onClick: function (event) {
                    event.preventDefault();
                    event.stopPropagation();
                    _this.props.deleteFile.apply(_this.props.index);
                }, className: "rcd_deleteButton" })));
    };
    return File;
}(React.Component));
exports.File = File;
//# sourceMappingURL=File.js.map