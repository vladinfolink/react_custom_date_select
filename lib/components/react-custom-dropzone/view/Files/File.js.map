{"version":3,"file":"File.js","sourceRoot":"","sources":["../../../../../src/components/react-custom-dropzone/view/Files/File.tsx"],"names":[],"mappings":";;;AAAA,mDAA8B;AAY9B;IAA0B,gCAA6B;IAErD,cAAY,KAAU;QAAtB,YACE,kBAAM,KAAK,CAAC,SAMb;QALC,KAAI,CAAC,KAAK,GAAG;YACX,WAAW,EAAE,EAAE;SAChB,CAAA;QACD,KAAI,CAAC,eAAe,GAAG,KAAI,CAAC,eAAe,CAAC,IAAI,CAAC,KAAI,CAAC,CAAC;QACvD,KAAI,CAAC,WAAW,GAAG,KAAI,CAAC,WAAW,CAAC,IAAI,CAAC,KAAI,CAAC,CAAC;;IACjD,CAAC;IAEI,gCAAiB,GAAvB;;;gBACG,IAAG,IAAI,CAAC,KAAK,CAAC,UAAU;oBAAE,IAAI,CAAC,WAAW,EAAE,CAAC;;;;KAC9C;IAEA,8BAAe,GAAf;QACC,OAAO,IAAI,CAAC,KAAK,CAAC,IAAI,CAAC,IAAI,CAAC,KAAK,CAAC,GAAG,CAAC,CAAC,CAAC,CAAC,KAAK,OAAO,CAAC;IACxD,CAAC;IAEA,0BAAW,GAAX;QAAA,iBAaA;QAZC,CAAC;;;;;;wBACO,MAAM,GAAG,IAAK,MAAc,CAAC,UAAU,EAAE,CAAC;wBAChD,WAAM,MAAM,CAAC,aAAa,CAAC,IAAI,CAAC,KAAK,CAAC,IAAI,CAAC,EAAA;;wBAA3C,SAA2C,CAAC;wBAC5C,MAAM,CAAC,MAAM,GAAG;4BACd,KAAI,CAAC,QAAQ,CAAC;gCACZ,IAAM,MAAM,GAAQ,MAAM,CAAC,MAAM,CAAC;gCAClC,OAAO;oCACL,WAAW,EAAE,MAAM;iCACpB,CAAA;4BACH,CAAC,CAAC,CAAA;wBACJ,CAAC,CAAC;;;;aACH,CAAC,EAAE,CAAC;IACP,CAAC;IAED,qBAAM,GAAN;QAAA,iBAuBC;QAtBG,OAAO,CACT,oBAAC,KAAK,CAAC,QAAQ;YACZ,IAAI,CAAC,KAAK,CAAC,UAAU,IAAI,IAAI,CAAC,eAAe,EAAE,IAAI,CAClD,6BACE,SAAS,EAAC,aAAa,EACvB,GAAG,EAAE,IAAI,CAAC,KAAK,CAAC,WAAW,EAC3B,GAAG,EAAE,IAAI,CAAC,KAAK,CAAC,IAAI,CAAC,IAAI,GACzB,CACH;YACD,8BAAM,SAAS,EAAC,qBAAqB,IAAE,IAAI,CAAC,KAAK,CAAC,IAAI,CAAC,IAAI,CAAQ;YACnE,gCACE,OAAO,EAAE,UAAC,KAAU;oBAClB,KAAK,CAAC,cAAc,EAAE,CAAC;oBACvB,KAAK,CAAC,eAAe,EAAE,CAAC;oBACxB,KAAI,CAAC,KAAK,CAAC,UAAU,CAAC,KAAK,CAAC,KAAI,CAAC,KAAK,CAAC,KAAK,CAAC,CAAA;gBAE/C,CAAC,EAED,SAAS,EAAC,kBAAkB,GAC5B,CACa,CAClB,CAAC;IACF,CAAC;IACH,WAAC;AAAD,CAAC,AA1DD,CAA0B,KAAK,CAAC,SAAS,GA0DxC;AA1DY,oBAAI"}