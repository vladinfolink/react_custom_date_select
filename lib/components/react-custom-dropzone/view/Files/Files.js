"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var React = tslib_1.__importStar(require("react"));
var File_1 = require("./File");
exports.Files = function (props) {
    var fileItems = props.files.map(function (file, index) {
        return (React.createElement("div", { className: "rcd_fileName", key: file.name + String(index) },
            React.createElement(File_1.File, { index: index, file: file, renderPics: props.renderPics, deleteFile: props.deleteFile })));
    });
    return (React.createElement("div", { className: "reactCustomDropzoneFiles" },
        React.createElement("div", { className: "allowedFilesContainer" },
            !!props.files.length,
            " ",
            fileItems)));
};
//# sourceMappingURL=Files.js.map