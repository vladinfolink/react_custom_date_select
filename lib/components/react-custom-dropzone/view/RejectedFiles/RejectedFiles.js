"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var React = tslib_1.__importStar(require("react"));
exports.RejectedFiles = function (props) {
    var rejectedFileItems = props.rejectedFiles.map(function (rejectedFile, index) {
        return (React.createElement("div", { className: "rcd_fileName", key: rejectedFile.name + String(index) },
            React.createElement("span", { className: "rcd_fileNameContent" }, rejectedFile.name)));
    });
    return (React.createElement("div", { className: "reactCustomDropzoneRejectedFiles" },
        React.createElement("div", { className: "rejectedText" }, "Rejected files:"),
        React.createElement("div", { className: "rejectedFilesContainer" }, rejectedFileItems)));
};
//# sourceMappingURL=RejectedFiles.js.map