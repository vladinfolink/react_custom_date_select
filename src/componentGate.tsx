import { ReactCustomDateSelect } from "./components/ReactCustomDateSelect";

export { ReactCustomDateSelect };
