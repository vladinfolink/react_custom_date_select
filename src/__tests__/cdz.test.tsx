import * as React from 'react';

import { CustomDropZone } from '../componentGate';
import { CustomDropZoneContainer } from "../components/react-custom-dropzone/containers/CustomDropzoneContainer/CustomDropzoneContainer";
import { Files } from "../components/react-custom-dropzone/view/Files/Files";
import { RejectedFiles } from "../components/react-custom-dropzone/view/RejectedFiles/RejectedFiles";

import { shallow } from 'enzyme';
import { create } from "react-test-renderer";

interface IProps {
  renderPics: boolean,
  allowClick: boolean,
  displayFiles: boolean,
  displayRejectedFiles: boolean,
  fileCountLimit: number,
  rejectedFileCountLimit: number,
  dropText: string,
  handleFiles: any,
  fileTypesAllowed: string[],
  maxFileSize: number
}
interface IState {
  files: any[],
  rejectedFiles: any[]
}

export class TestCustomDropZone extends React.Component<IProps, IState> {

  static defaultProps = {
    renderPics: true,
    allowClick: true,
    displayFiles: true,
    displayRejectedFiles: true,
    fileCountLimit: Infinity,
    rejectedFileCountLimit: Infinity,
    dropText: '',
    handleFiles: (params: any) => {
      console.log(params);
    },
    fileTypesAllowed: [],
    maxFileSize: Infinity
  }

  constructor(props: any) {
    super(props);
    this.state = {
      files: [],
      rejectedFiles: []
    };
    this.updateState = this.updateState.bind(this);
    this.shouldDisplayFiles = this.shouldDisplayFiles.bind(this);
  }

  private handleDrop = (fileItems: any[]) => {
    const typesAllowed: any[] = [...this.props.fileTypesAllowed];
    const maxFileSize: number = this.props.maxFileSize;
    const { files, rejectedFiles } = this.state;

    const inboundFiles = !!typesAllowed.length ? Array.from(fileItems).reduce(
      function (ini, item) {
        return (typesAllowed.indexOf(item.type) !== -1 && item.size <= maxFileSize)
          ? { ...ini, files: [...ini.files, item] }
          : { ...ini, rejectedFiles: [...ini.rejectedFiles, item] };
      },
      { files: [], rejectedFiles: [] }
    ) : {
        files: Array.from(fileItems), rejectedFiles: []
      };

    for (const iterator of inboundFiles.files) {
      if (files.length < this.props.fileCountLimit) {
        files.push(iterator);
      }
    }

    for (const iterator of inboundFiles.rejectedFiles) {
      if (rejectedFiles.length < this.props.rejectedFileCountLimit) {
        rejectedFiles.push(iterator);
      }
    }
    return this.updateState({ files, rejectedFiles });
  };

  private updateState(combinedFilesObj: { files: any, rejectedFiles: any }) {
    const { files, rejectedFiles } = combinedFilesObj;
    this.setState(
      function () {
        return {
          files,
          rejectedFiles
        };
      },
      () => {
        this.props.handleFiles({ ...this.state });
      }
    );
  }

  private deleteFile = (fileIndex: number) => {
    const files = [...this.state.files];
    files.splice(fileIndex, 1);
    this.setState(
      function () {
        return { files };
      },
      () => {
        this.props.handleFiles({ ...this.state });
      }
    );
  };

  private shouldDisplayFiles() {
    return this.props.displayFiles
  }

  render() {
    const dropText = !this.state.files.length ? this.props.dropText : null;

    return (
      <React.Fragment>
        <CustomDropZoneContainer
          dropText={dropText}
          handleDrop={this.handleDrop}
          allowClick={this.props.allowClick}
        >
          {this.shouldDisplayFiles() && (
            <Files
              renderPics={this.props.renderPics}
              files={this.state.files}
              deleteFile={this.deleteFile}
            />
          )}
          {
            !!this.state.rejectedFiles.length &&

            this.props.displayRejectedFiles && (
              <RejectedFiles rejectedFiles={this.state.rejectedFiles} />
            )}
        </CustomDropZoneContainer>
      </React.Fragment>
    );
  }
}


it('render CustomDropZone', function () {
  expect(shallow(<CustomDropZone />).length).toEqual(1)
});

describe("TestCustomDropZone component", () => {
  test("any", () => {
    const component = (create as any)(<TestCustomDropZone />);
    const instance = component.getInstance();
    expect(instance.props.dropText).toBe("");
    expect(shallow(<TestCustomDropZone />)).toMatchSnapshot()
  });
});