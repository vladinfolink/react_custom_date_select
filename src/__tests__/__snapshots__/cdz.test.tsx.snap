// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestCustomDropZone component any 1`] = `
<Fragment>
  <CustomDropZoneContainer
    allowClick={true}
    dropText=""
    handleDrop={[Function]}
  >
    <Component
      deleteFile={[Function]}
      files={Array []}
      renderPics={true}
    />
  </CustomDropZoneContainer>
</Fragment>
`;
